### ISSUE TITLE
Create route for getting alerts

### ISSUE THREAD

**Issue by benburwell** [MCEMS/api#25]
We need to create a route that will allow the dispatch console to pull data about current and recent calls from Active911.
1. Add the `active911` and `redis` modules as dependencies to `package.json`
2. Add `heroku-redis` to the `addons` in `app.json`
3. Add the [redis image](https://hub.docker.com/_/redis/) as a new container definition in `docker-compose.yml`. Refer to the configuration for Postgres as an example.
   - Redis uses TCP port 6379
   - Give the `web` container an environment variable named `REDIS_URL` that is the URL. This mirrors the production environment, the Heroku Redis addon exposes a `REDIS_URL` that the application can connect to.
4. Create a file `lib/redis.coffee`. This file will be a Hapi plugin that exposes a redis connection. You will need to create a function `module.exports.register` that uses the `redis` module and connects to `process.env.REDIS_URL`. Then, `server.expose` the created client as `redis`. This will allow other files to access the Redis connection by referring to `server.plugins['redis']`. You will also need to create `module.exports.register.attributes`, see `lib/bookshelf/index.coffee` for an example.
5. Create a file `lib/active911.coffee`. This file will be a Hapi plugin that exposes an `Active911.RefreshClient` in a similar way to the redis plugin. You'll create a new client passing in `process.env.ACTIVE911_REFRESH_TOKEN`, and `server.expose` the client as `active911`.
6. Edit `lib/server.coffee`. Before `require`ing the routes, you will need to also register your two new plugins. Call `server.register require('./redis')` and `server.register require('./active911')`, similar to how bookshelf is being required.
7. Create a file at `lib/api/v1/active911/Alert.coffee`. This file should have `module.exports` defined as a function on the server object that adds a route with the following details (the handler will be described below)
   - Method: GET
   - Path: `/api/v1/active911/alert`
   - Auth scopes: `send_alert`

The JSON returned from the new endpoint should look like this:

``` javascript
[
  {
    "id": "<id key from active911 alert>",
    "description": "<description field from active911>",
    "details": "<details field from active911>",
    "place": "<place field from active911>",
    "address": "<address field from active911>",
    "timestamp": "<received field from active911>",
    "responses": [
      {                                                                                                                                        
        "name": "<device name from active911>",
        "response": "<response field from active911>",
        "timestamp": "<timestamp field from active911>"
      }
    ]
  }
]
```

In order to fetch this data from the Active911 API, use the `server.plugins['active911']` client that you previously created. Active911 will return some data that we're not really interested in, as well as other data that isn't formatted in a very convenient way for our purposes. Specifically, additional requests will be necessary to get the name associated with each device that is responding.

Here is the general flow for building the response:
1. Call `client.getAlerts()`. This will asynchronously return (via a promise) an array of alert references that contain an `id` and a `url`.
2. For each alert ID that you get, you will need to call `client.getAlert(id)` to get more details about the alert.
3. For each response in the `responses` array, you will need to call `client.getDevice(id)` to see what the name of the device is.

_see [this page](http://wiki.active911.com/wiki/index.php/Accessing_the_API) for details about the exact data you will get back_

Now, you should have all the information necessary to build up the actual JSON that you'll be returning. As you can imagine, doing all of this each time we need to render the page would result in way too many HTTP callouts and be very slow. This is where Redis comes in. Redis is a "data structure server" that we will be using as a cache.

So before you use the Active911 client to make a request, check to see if we already have the data cached by pinging Redis. If it's there, just use that data. Otherwise, make the HTTP callout using the Active911 client and then store the results in Redis to use for subsequent requests. You'll also use the [EXPIRE command](http://redis.io/commands/expire) to tell Redis when it should remove the key and force the client to refresh it from the server.

Here are the keys that should be stored in Redis, along with their types and expirations:
- `active911:alerts`
  - Type: Set
  - Expiration: 30 seconds
  - Holds: a set of string IDs of alerts returned by `getAlerts()`
- `active911:alert:<ID>`
  - Type: String
  - Expiration: 30 seconds
  - Holds: unfortunately, a string-serialized JSON object that is the response from `getAlert(<ID>)`. This is because `responses`, which is what we really need, is an anonymous collection; there is no way to address a particular response. Furthermore, since we'd have to requery the API for the whole alert anyway to see if any responses are different, there's really no advantage to storing the general alert data separately from the responses at this point.
- `active911:device_name:<ID>`
  - Type: String
  - Expiration: 1 day
  - Holds: the `name` returned from `getDevice(<ID>)`.

Please let me know if there is anything you need me to clarify or that you have questions about!

**Comment by benburwell** [comment]
I just published [`active911` v1.1.1](https://www.npmjs.com/package/active911) which allows you to get alerts for the past _k_ minutes/days. When you invoke `getAlerts`, use:

``` javascript
getAlerts({ alert_days: 1 })
```

to get alerts from the past day.

**Comment by benburwell** [comment]
Resolved in 82455a8
